File: 백준/Silver/11403. 경로 찾기/경로 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;
        int N = Integer.parseInt(br.readLine());
        int[][] graph = new int[N][N];
        for(int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            for(int j = 0; j < N; j++) {
                graph[i][j] = Integer.parseInt(st.nextToken());
            }
        }
        for(int k = 0; k < N; k++) {
            for(int i = 0; i < N; i++) {
                for(int j = 0; j < N; j++) {
                    if(graph[i][k] == 1 && graph[k][j] == 1) {
                        graph[i][j] = 1;
                    }
                }
            }
        }
        for(int i = 0; i < N; i++) {
            for(int j = 0; j < N; j++) {
                sb.append(graph[i][j]);
                if(j != N - 1) sb.append(" ");
            }
            sb.append("\n");
        }
        System.out.println(sb.toString().trim());
    }
}

File: 백준/Silver/11403. 경로 찾기/경로 찾기.js
let [N, ...graph] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n').map(line => line.trim());
N = Number(N);
graph = graph.map(line => line.split(' ').map(Number));
for(let k = 0; k < N; k++) {
  for(let i = 0; i < N; i++) {
    for(let j = 0; j < N; j++) {
      if(graph[i][k] && graph[k][j]) {
        graph[i][j] = 1;
      }
    }
  }
}
console.log(graph.map(line => line.join(' ')).join('\n'));